src/be_db_reward.erl

```diff
@@ -71,17 +77,68 @@ load_block(Conn, _Hash, Block, _Sync, _Ledger, State = #state{}) ->
     {ok, State}.
 
 -type reward_map() :: #{
-    {Account :: libp2p_crypto:pubkey_bin(), Gateway :: libp2p_crypto:pubkey_bin()} =>
+    {Account :: libp2p_crypto:pubkey_bin(), Gateway :: libp2p_crypto:pubkey_bin() | undefined} =>
         Reward :: pos_integer()
 }.
 
--spec collect_rewards(blockchain_txn_reward_v1:rewards(), reward_map()) -> reward_map().
-collect_rewards([], RewardMap) ->
+collect_rewards(blockchain_txn_rewards_v1, _Chain, Txn, RewardMap) ->
+    collect_v1_rewards(blockchain_txn_rewards_v1:rewards(Txn), RewardMap);
+collect_rewards(blockchain_txn_rewards_v2, Chain, Txn, RewardMap) ->
+    Start = blockchain_txn_rewards_v2:start_epoch(Txn),
+    End = blockchain_txn_rewards_v2:end_epoch(Txn),
+    {ok, Ledger} = blockchain:ledger_at(End, Chain),
```

Comment:
I'm assuming this is never expected to blow up because of the 1:1 sync that ETL does? `ledger_at` usually blows up if you're looking behind 50 blocks iirc.

Reply:
indeed.. I do this because this is what calculate_rewards_metadata in rewards_v2 does, and we're in lockstep with that, so if it'd break ETL it would also break the chain